multi-services/eks-multi-services-dispatcher.c

```diff
@@ -0,0 +1,166 @@
+/* Copyright 2016 Endless Mobile, Inc. */
+
+#include <gio/gio.h>
+#include <glib.h>
+
+#include <string.h>
+
+static char *services_version;
+
+static GOptionEntry entries[] = {
+  { "services-version", 's', 0, G_OPTION_ARG_STRING, &services_version, "The eks-search-provider version to use", "VERSION" },
+  { NULL }
+};
+
+static GStrv
+envp_search (GStrv       envp,
+             const char *variable)
+{
+  gsize variable_len = strlen (variable);
+
+  for (; *envp != NULL; ++envp)
+    {
+      if (strncmp (*envp, variable, variable_len) == 0)
+        return envp;
+    }
+
+  return NULL;
+}
+
+static void
+insert_paths_to_env_var (GStrv               env,
+                         const char         *variable,
+                         const char * const *paths)
+{
+  GStrv envp_index = envp_search (env, variable);
+  g_autofree char *join_envp = NULL;
+
+  if (envp_index == NULL)
+    return;
+
+  join_envp = g_strjoinv (":", (GStrv) paths);
+
+  /* Free the string already at envp_index */
+  g_free (*envp_index);
+
+  *envp_index = g_strdup_printf ("%s=%s", variable, join_envp);
+}
+
+static GSubprocess *
+spawnv_with_appended_paths_and_fds (const char * const  *argv,
+                                    const char * const  *executable_paths,
+                                    const char * const  *ld_library_paths,
+                                    const char * const  *xdg_data_dirs,
+                                    GError             **error)
+{
+  g_autoptr(GSubprocessLauncher) launcher = g_subprocess_launcher_new (G_SUBPROCESS_FLAGS_INHERIT_FDS);
+  g_auto(GStrv) env = g_listenv ();
```

Comment:
Looks like it would be a bit more convenient to use `g_get_environ()` and `g_environ_get/setenv()`